### PR TITLE
Add definition for shallow-equals

### PR DESCRIPTION
**Pull request by rsolomon**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/shallow-equals/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Type definitions for shallow-equals v1.0.0
+// Project: https://github.com/hughsk/shallow-equals
+// Definitions by: Ross Solomon <https://github.com/rsolomon>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = shallow_equals;
+
+declare function shallow_equals<A, B>(
```
Comment (rsolomon) [inline types/shallow-equals/index.d.ts]: I considered making both `a` and `b` the same type `T`, but figured that might be too strict for some use cases.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -3968,6 +3968,7 @@
 /types/set-value/                               @DanielRosenwasser
 /types/settings/                                @shreyjain1994
 /types/sha1/                                    @arcdev1
+/types/shallow-equals/                          @rsolomon
```
Comment (uniqueiniquity) [inline .github/CODEOWNERS]: The codeowners file will be updated automatically on publish, so please remove this edit.

File: types/shallow-equals/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  a?: A,
+  b?: B,
+  compare?: (objA?: A, objB?: B) => boolean,
```
Comment (uniqueiniquity) [inline types/shallow-equals/index.d.ts]: As discussed [here](http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html) under "Optional Parameters in Callbacks", please don't make `objA` and `objB` optional.

File: types/shallow-equals/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": ["es6"],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (uniqueiniquity) [inline types/shallow-equals/tsconfig.json]: Is there a particular reason you have `strictNullChecks` disabled?

File: types/shallow-equals/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+): boolean;
+
+declare namespace shallow_equals {}
```
Comment (uniqueiniquity) [inline types/shallow-equals/index.d.ts]: This is not necessary; you can remove it.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/shallow-equals/tsconfig.json (hunk 4 above)
Comment (rsolomon) [reply types/shallow-equals/tsconfig.json]: Nope, my bad!

File: types/shallow-equals/index.d.ts (hunk 3 above)
Comment (rsolomon) [reply types/shallow-equals/index.d.ts]: TIL

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Comment by rsolomon** [comment]
@RyanCavanaugh @andy-ms Heads up this package isn't yet available in npm after 24 hours. It may be because it still has the "Revision needed" label. Thanks in advance!

**Pull request merged**